### ISSUE TITLE
np.intp can't necessarily hold GPU pointers

### ISSUE THREAD

**Issue by bmerry** [inducer/pycuda#56]
When doing development on a 32-bit ARM system I was getting this error (not on all programs though):

```
  File "/home/ubuntu/work/sdp/env/local/lib/python2.7/site-packages/pycuda/driver.py", line 377, in function_call
    handlers, arg_buf = _build_arg_buf(args)
  File "/home/ubuntu/work/sdp/env/local/lib/python2.7/site-packages/pycuda/driver.py", line 150, in _build_arg_buf
    gpudata = np.intp(arg.gpudata)
OverflowError: Python int too large to convert to C long
```

Changing `np.intp` to `np.uintp` made things work for me. Looking at mem_obj_to_long, it seems to always interpret pointers as unsigned, so hopefully this is safe on x86 as well.

**Comment by dtaviation** [comment]
I find that I have this problem running an example:

$ python demo_struct.py
Traceback (most recent call last):
  File "demo_struct.py", line 31, in <module>
    array1 = DoubleOpStruct(numpy.array([1, 2, 3], dtype=numpy.float32), struct_arr)
  File "demo_struct.py", line 23, in __init__
    numpy.getbuffer(numpy.intp(int(self.data))))
OverflowError: Python int too large to convert to C long


$ grep intp /usr/local/lib/python2.7/dist-packages/pycuda-2016.1-py2.7-linux-armv7l.egg/pycuda/driver.py
                    gpudata = np.uintp(arg.gpudata)
                func.arg_format += np.dtype(np.uintp).char
                func.arg_format += np.dtype(np.uintp).char

**Comment by dtaviation** [comment]
More information about my system:

NVidia Jetson Tk1, armv7 32-bit.

$ cat /etc/os-release 
NAME="Ubuntu"
VERSION="14.04.4 LTS, Trusty Tahr"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 14.04.4 LTS"
VERSION_ID="14.04"
HOME_URL="http://www.ubuntu.com/"
SUPPORT_URL="http://help.ubuntu.com/"
BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"

$ uname -a
Linux tk2 3.10.40-gdacac96 #1 SMP PREEMPT Thu Jun 25 15:25:11 PDT 2015 armv7l armv7l armv7l GNU/Linux

**Comment by inducer** [comment]
Struct example fixed in 9e08f58.
